### PR TITLE
Added documentation build check to CI

### PR DESCRIPTION
**Pull request by HU90m**

### CODE REVIEW

**Review by marnovandermaas — COMMENTED** [review]
The change looks good to me. Do you have this action running anywhere?

File: .github/workflows/mdbook.yml [hunk 1]
```diff
@@ -21,29 +21,29 @@ concurrency:
...
+      with:
+        extra_nix_config: |
+          substituters = https://nix-cache.lowrisc.org/public/ https://cache.nixos.org/
+          trusted-public-keys = nix-cache.lowrisc.org-public-1:O6JLD0yXzaJDPiQW1meVu32JIDViuaPtGDfjlOopU7o= cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
+
+    - name: Build Documentation
+      run: |
+        nix build .#sonata-documentation
+        # copy out of /nix/store with all symlinks dereferenced and with default permissions
+        cp -LR --no-preserve=mode,ownership result build
```
Comment (nbdd0121) [inline .github/workflows/mdbook.yml:40]: You probably also want to ignore timestamps, although I am not sure if GitHub pages action actually care or not.

Comment (HU90m) [reply .github/workflows/mdbook.yml]: Removed ownership

**Review by nbdd0121 — APPROVED** [review]

**Comment by marnovandermaas** [comment]
Action now failing: https://github.com/lowRISC/sonata-system/actions/runs/9549968503

**Pull request merged**